src/engine/source/kvdb/src/kvdbManager.cpp

```diff
@@ -1,412 +1,346 @@
+    }
 
-        json::Json jKv;
-        try
-        {
-            jKv = json::Json {contents.c_str()};
-        }
-        catch (const std::exception& e)
+    auto it = m_mapCFHandles.find(name);
+    if (it != m_mapCFHandles.end())
+    {
+        auto cfHandle = it->second;
+        auto opStatus = m_pRocksDB->DropColumnFamily(cfHandle);
+        if (opStatus.ok())
         {
-            return base::Error {fmt::format("An error occurred while parsing the JSON file '{}'", path.c_str())};
+            m_mapCFHandles.erase(it);
         }
-
-        if (!jKv.isObject())
+        else
         {
             return base::Error {
-                fmt::format("An error occurred while parsing the JSON file '{}': JSON is not an object", path.c_str())};
+                fmt::format("Could not remove the DB {}. RocksDB Status: {}", name, opStatus.ToString())};
         }
-
-        entries = jKv.getObject().value();
     }
-
-    // Check if the database exists
-    if (exist(dbName))
-    {
-        return base::Error {fmt::format("Database '{}' already exists", dbName)};
-    }
-
-    // Create the database
-    auto kvdb = getHandler(dbName, true);
-    if (std::holds_alternative<base::Error>(kvdb))
-    {
-        return std::get<base::Error>(kvdb);
-    }
-
-    for (const auto& [key, value] : entries)
+    else
     {
-        writeKey(dbName, key, value); // TODO check error
+        return base::Error {fmt::format("The DB not exists.")};
     }
 
     return std::nullopt;
 }
 
-std::variant<json::Json, base::Error> KVDBManager::jDumpDB(const std::string& name)
+std::optional<base::Error> KVDBManager::loadDBFromFile(const std::string& name, const std::string& path)
 {
-    auto handle = getHandler(name);
-    if (std::holds_alternative<base::Error>(handle))
+    std::vector<std::tuple<std::string, json::Json>> entries {};
+    rocksdb::ColumnFamilyHandle* cfHandle {nullptr};
+
+    if (m_mapCFHandles.count(name))
     {
-        return std::get<base::Error>(handle);
+        cfHandle = m_mapCFHandles[name];
     }
-    auto& kvdb = std::get<KVDBHandle>(handle);
 
-    return kvdb->jDump();
-}
-
-std::variant<std::unordered_map<std::string, std::string>, base::Error> KVDBManager::rDumpDB(const std::string& name)
-{
-    auto handle = getHandler(name);
-    if (std::holds_alternative<base::Error>(handle))
+    if (!cfHandle)
     {
-        return std::get<base::Error>(handle);
+        return base::Error {fmt::format("The DB not exists.")};
     }
-    auto& kvdb = std::get<KVDBHandle>(handle);
 
-    return kvdb->rDump();
-}
-
-std::optional<base::Error> KVDBManager::writeRaw(const std::string& name,
-                                                 const std::string& key,
-                                                 const std::string value)
-{
-    auto handle = getHandler(name);
-    if (std::holds_alternative<base::Error>(handle))
+    // TODO: to improve
+    if (std::empty(path))
```

Comment:
```suggestion
    if (path.empty())
```
Same recommendation applies to other occurrencies